crates/oxc_linter/src/rules/typescript/prefer_function_type.rs

```diff
@@ -0,0 +1,743 @@
+use oxc_ast::{
+    ast::{ExportDefaultDeclarationKind, Expression, TSInterfaceDeclaration, TSSignature, TSType},
+    AstKind, CommentKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, fixer::Fix, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(prefer-function-type): Enforce using function types instead of interfaces with call signatures.")]
+#[diagnostic(
+    severity(warning),
+    help("See https://typescript-eslint.io/rules/prefer-function-type for documentation.")
```

Comment:
Can can improve the help message by saying what to do, e.g.

The function type form `() => string` is generally preferred when possible for being more succinct.

Reply:
![image](https://github.com/oxc-project/oxc/assets/51067792/2fda9d84-cc34-4cb7-bba8-4dc4c30852a9)